app/routes.py

```diff
@@ -1 +1,192 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint
+from os import abort, environ
+import os
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, abort, make_response, request
+from datetime import datetime
+import requests
+from dotenv import load_dotenv
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals",__name__, url_prefix="/goals")
+
+def get_valid_item_by_id(model, id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({'msg': f"Invalid id '{id}'"}, 400))
+
+    item = model.query.get(id)
+
+    return item if item else abort(make_response({'msg': f"No {model.__name__} with id {id}"}, 404))
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if not "title" in request_body or not "description" in request_body:
+        abort (make_response({"details": "Invalid data"}, 400))
+
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"])
+                    # completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task" : new_task.to_result()}, 201
+
+@tasks_bp.route("<task_id>/mark_complete", methods=['PATCH'])
+def mark_task(task_id):
+    task = get_valid_item_by_id(Task,task_id)
+    request_body = request.get_json()
+
+    task.completed_at = datetime.now()
+    
+    db.session.commit()
+
+    slack_api = "https://slack.com/api/chat.postMessage"
```

Comment:
In the future, it might be nice to refactor `slack_api` into a [constant variable](https://realpython.com/python-constants/#user-defined-constants) and renaming it to `SLACK_API`. This is because this variable contains a value that we don't think will ever change within this scope